libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -215,15 +212,28 @@ class NavigationView : ConstraintLayout {
     ) : super(context, attrs) {
         this.mapInitOptions = mapInitializationOptions
         this.navigationViewOptions = navigationViewOptions
-        this.lifeCycleOwner = context as? LifecycleOwner ?: throw LifeCycleOwnerNotFoundException()
-        this.lifeCycleOwner.lifecycle.addObserver(lifecycleObserver)
+
+        val hostingLifecycleOwner = context.unwrapIfNeeded() as? LifecycleOwner
+            ?: throw LifeCycleOwnerNotFoundException()
+        this.viewLifecycleOwner = ViewLifecycleOwner(hostingLifecycleOwner)
```

Comment:
Considering that Fragment and Activity inherit from the `LifecycleOwner` interface. I think it is more consistent to make the `NavigationView` inherit from `LifecycleOwner` (you think about it, it is indeed a lifecycle owner after this change).

A delegating class that allows any View to easily become a `LifecycleOwner`, would still be good.

Reply:
Done, the only thing I'm worried about is that now we make the lifecycle of the `NavigationView` part of the public API. Not entirely sure that we want to do that but we have enough time to keep iterating and see if this might cause any problems.

Reply:
Resolving this thread, made another comment on this code https://github.com/mapbox/mapbox-navigation-android/pull/5332#discussion_r780575597